pkg/compare/parsing.go

```diff
@@ -34,14 +37,72 @@ const (
 	Optional ComponentType = "Optional"
 )
 
+const (
+	fieldsToOmitBuiltInOverwritten = `fieldsToOmit.Map contains the key "%s", this will be overwritten with default values`
+	fieldsToOmitDefaultNotFound    = `fieldsToOmit's defaultKey "%s" not found in items`
+	fieldsToOmitRefsNotFound       = `skipping fieldsToOmitRefs entry: "%s" not found it fieldsToOmit Items`
+)
+
+type FieldsToOmitConfig struct {
+	DefaultKey string `json:"defaultKey,omitempty"`
+}
+
+type FieldsToOmit struct {
+	Config         FieldsToOmitConfig  `json:"config,omitempty"`
+	Items          map[string][]string `json:"items,omitempty"`
+	processedItems map[string][]Path
+}
+
+// Setup FieldsToOmit to be used by setting defaults
+// and processing the item strings into paths
+func (toOmit *FieldsToOmit) process() error {
+	if toOmit.Items == nil {
+		toOmit.Items = make(map[string][]string)
+	}
+
+	if _, ok := toOmit.Items[builtInPathsKey]; ok {
+		klog.Warningf(fieldsToOmitBuiltInOverwritten, builtInPathsKey)
+	}
+
+	toOmit.Items[builtInPathsKey] = builtInPaths
+
+	if toOmit.Config.DefaultKey == "" {
+		toOmit.Config.DefaultKey = builtInPathsKey
+	}
+
+	if _, ok := toOmit.Items[toOmit.Config.DefaultKey]; !ok {
+		return fmt.Errorf(fieldsToOmitDefaultNotFound, toOmit.Config.DefaultKey)
+	}
+
+	toOmit.processedItems = make(map[string][]Path)
+	for key, pathsArray := range toOmit.Items {
+		processedPaths := make([]Path, 0)
+		for _, path := range pathsArray {
+			p, err := NewPath(path)
+			if err != nil {
+				klog.Errorf("skipping path: %s", err)
+				continue
+			}
+			processedPaths = append(processedPaths, p)
+		}
+		if len(processedPaths) == 0 {
+			klog.Errorf("skipping key: no paths in key %s", key)
```

Comment:
lets return errors instead of using klog,
if there is a problem it should not skip, this isnt a dynamic problem, its a problem in the ref structure, this can not be a vaild ref, it will make this pop for the user each time. 
(and are you sure it skips?  i remember the klog.errorf exits, anyway i think  it should not skip ) 
if you want to return all the errors from this function  i think its better to use error.join and then return it so the user gets all the errors at the same run of the command.

Reply:
I'll take a look but yes thats fair it should fail.